packages/plexus/README.md

```diff
@@ -1,13 +1,914 @@
+
+The type for `setOnNode` is similar to that of `setOnContainer` in that the value can also be either an object of props, a factory function, or an array of either.
+
+```tsx
+type TNodePropsSetter = Record<string, unknown> | TNodesPropsFn | (TNodesPropsFn | Record<string, unknown>)[];
+
+type TNodesPropsFn = (
+  layoutVertex: TLayoutVertex | null,
+  utils: TRendererUtils
+) => Record<string, unknown> | null;
+```
+
+The type for the `renderNode` field.
+
+```tsx
+type TRenderNodeFn = (layoutVertex: TLayoutVertex, utils: TRendererUtils) => React.ReactNode;
+```
+
+#### Edges layer
+
+Any number of edges layers can be configured for a `LayoutDigraph` (TODO: rename). Edges layers are more restrictive (or less mature) than nodes layers, at present:
+
+- The `layerType` of edges layers must be `"svg"`
+- Edges layers do not afford a `renderEdge` equivalent to the `renderNode`.
+
+Thus, edges are less configurable than nodes (for now). If you need additional functionality, please [file a ticket](https://github.com/jaegertracing/jaeger-ui/issues/new?labels=plexus&title=Plexus+edge+enhancements).
+
+The builtin renderer for edges draws a path based on the `pathPoints` field of the `TLayoutEdge` being rendered.
+
+In addition to the common layer configuration fields, the following fields are also available:
+
+| Name | Type and description |
+| :-- | :-- |
+| edges | `true` |
+|  | **Required**<br>Indicates the layer will render edges.<br>&nbsp; |
+| layerType | `"svg"` |
+|  | **Required**<br>Edges must be SVG layers.<br>&nbsp; |
+| markerEndId | `string` |
+|  | This field should refer to an element to use as the [`marker-end`](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/marker-end) for the edge `<path>`. A typical scenario matches the `markerEndId` to the `localId` of a `TDefEntry`. Each of the `localId` and `markerEndId` are passed through the `getLocalId()` util (TODO: reaname the util) to generate an ID unique within the document.<br>&nbsp; |
+| markerStartId | `string` |
+|  | The [`marker-start`](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/marker-start) equivalent of the `markerEndId` field.<br>&nbsp; |
+| setOnEdge | `TEdgePropsSetter` _See below for details on this type._ |
+|  | Allows props to be defined or generated for the `<path>`. Unlike nodes, edges are not wrapped in a container. So, the resultant props are applied directly to the `<path>`.<br>&nbsp; |
+| defs | `TDefEntry[]` |
+|  | For edges, this is generally used to define arrows (or other markers) in order to indicate directionality. See [`TDefEntry`](https://github.com/tiffon/jaeger-ui/wiki/_new#tdefentry) for additional details.<br><br>The default functionality of a `TDefEntry` is suitable to be the `markerEnd` of an edge. To define an arrow head marker on an edge layer, simple set the `markerEndId` of the layer to the `localId` of the `TDefEntry`:<br><br>`{ defs: [{ localId: 'arrow' }], markerEndId: 'arrow', ...otherProps }`<br>&nbsp; |
```

Comment:
should this be: `defs: [{ localId: getLocalId('arrow') }]` or does the edges layer automatically localize these IDs?

Reply:
`getLocalId` is not in scope, at that time.

The edges layer automatically handles the naming.